src/context/legendPayloadContext.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { createContext, useCallback, useContext, useEffect, useState } from 'react';
+import { Payload as LegendPayload } from '../component/DefaultLegendContent';
+
+type PayloadSupplier = () => Array<LegendPayload>;
```

Comment:
nit nit: why the longform `Array<>` over `[]`

Reply:
Because I started with `ReadonlyArray` and then it turns out that Legend requires these to be mutable (because of the type definition, it doesn't actually mutate anything) so I deleted the `Readonly` part.

It behaves exactly the same anyway.

Reply:
yeah I know, all good I think I prefer `[]` but doesn't matter at all :p